### PR TITLE
Docs: Update template variable overview to contain the preferred variable syntax

### LINKED ISSUE #13744: Docs: document the new variable interpolation syntax
* `${varname}` 
* `${varname:csv}` csv formatting for multi valued variables
* `${varname:raw}` no escaping (very useful) 

Needs to be documented in the templating doc and probably mentioned in the templating section for each data source as well.

### PR DESCRIPTION
**Pull request by torkelo**
Fixes #13744

### CODE REVIEW

File: docs/sources/variables/templates-and-variables.md [hunk 1]
```diff
@@ -43,9 +43,9 @@ Panel titles and metric queries can refer to variables using two different synta
 - `$varname`
   This syntax is easy to read, but it does not allow you to use a variable in the middle of a word.
   **Example:** apps.frontend.$server.requests.count
-- `[[varname]]` 
-  **Note:** This syntax might be deprecated in the future.
-  Example: apps.frontend.[[server]].requests.count
+- `${var_name}` This syntax is needed when you want to interpolate a variable in the middle of an expression.
```
Comment (oddlittlebird) [inline docs/sources/variables/templates-and-variables.md:46]:
```suggestion
- `${var_name}` Use this syntax when you want to interpolate a variable in the middle of an expression. 
```

File: docs/sources/variables/templates-and-variables.md [hunk 2]
```diff
@@ -43,9 +43,9 @@ Panel titles and metric queries can refer to variables using two different synta
 - `$varname`
   This syntax is easy to read, but it does not allow you to use a variable in the middle of a word.
   **Example:** apps.frontend.$server.requests.count
-- `[[varname]]` 
-  **Note:** This syntax might be deprecated in the future.
-  Example: apps.frontend.[[server]].requests.count
+- `${var_name}` This syntax is needed when you want to interpolate a variable in the middle of an expression. 
+- `${var_name:<format>}` Gives you more control over how values are interpolated. Read [Advanced variable format options]({{< relref "advanced-variable-format-options.md" >}}) for more detail on all the formatting types.
```
Comment (oddlittlebird) [inline docs/sources/variables/templates-and-variables.md:47]:
```suggestion
- `${var_name:<format>}` This format gives you more control over how Grafana interpolates values. Refer to [Advanced variable format options]({{< relref "advanced-variable-format-options.md" >}}) for more detail on all the formatting types.
```

File: docs/sources/variables/templates-and-variables.md [hunk 3]
```diff
@@ -43,9 +43,9 @@ Panel titles and metric queries can refer to variables using two different synta
...
+- `[[varname]]` Deprecated old syntax, will be removed in a future release.
```
Comment (oddlittlebird) [inline docs/sources/variables/templates-and-variables.md:48]:
```suggestion
- `[[varname]]` Do not use. Deprecated old syntax, will be removed in a future release. 
```

Comment (oddlittlebird) [reply docs/sources/variables/templates-and-variables.md:48]: I'm told this syntax is necessary for certain use cases. @daniellee , can you confirm?

**Review by oddlittlebird — CHANGES_REQUESTED** [review]

File: docs/sources/variables/templates-and-variables.md (hunk 3 above)
Comment (torkelo) [reply docs/sources/variables/templates-and-variables.md:48]: no it's not, all those use cases is covered by  ${variable}

Comment (daniellee) [reply docs/sources/variables/templates-and-variables.md]: The issue is that there are thousands of dashboards using this the [[ ]] syntax so I don't think we will ever remove it. Or what do you say @torkelo ?

Comment (torkelo) [reply docs/sources/variables/templates-and-variables.md]: probably not :) maybe if we have a migration for it, but that is a bit tricky

**Pull request merged**